### PR TITLE
Deal with a range of system environment problems by docker and make it deploy easier

### PR DESCRIPTION
**Pull request by garysdevil**
Deal with a range of system environment problems by docker and make it deploy easier

### CODE REVIEW

File: docker.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+```bash
+repath=~/git/project/ # The folder path of repository bitcoinops.github.io
+
+cd ${repath}
+
+git clone https://github.com/bitcoinops/bitcoinops.github.io
+
+docker run -d --name bitcoinops -p 4000:4000 -v ${repath}/bitcoinops.github.io:/root/bitcoinops.github.io -w /root/bitcoinops.github.io ruby:2.6.4-stretch /bin/bash -c "bundle install && make preview"
+
+# open url 127.0.0.1:4000 in browser
```
Comment (bitschmidty) [inline docker.md:13]: This is not immediately available in the browser while the bundle install and make commands are running, so perhaps we can note the delay or provide a way for users to check on the build process (docker logs <container ID> ?) ?

File: docker.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+docker stop bitcoinops
+# edit file
```
Comment (bitschmidty) [inline docker.md:16]: Not sure if its notable here, but you dont need to stop in order to edit any of the repo files. The make preview will keep running and monitor for file changes and rebuild (in most cases).

**Review by bitschmidty — COMMENTED** [review]
Thanks @garysdevil for this addition!

ACK

I tested this successfully locally by following the commands in this markdown file. (sans upgrading from an old version of docker and an error when using a relative repath).

File: docker.md (hunk 1 above)
Comment (garysdevil) [reply docker.md:13]: Yeah, we can check whether is it deployed success by ``docker logs -f bitcoinops``, it deploys success if a user sees the log 'Server running'. I will update the file docker.md.
image

File: docker.md (hunk 2 above)
Comment (garysdevil) [reply docker.md]: > Not sure if its notable here, but you dont need to stop in order to edit any of the repo files. The make preview will keep running and monitor for file changes and rebuild (in most cases).

Thank your reminder.

Comment (garysdevil) [reply docker.md]: We must use absolute repath cause the docker mechanism.

Year, we can check whether is it deployed success by ``docker logs -f bitcoinops``.

I have optimized the file docker.md .

**Review by bitschmidty — COMMENTED** [review]
@garysdevil thanks for the updates, the build detected trailing spaces:

> 8:28:21 PM: docker.md:53: MD009 Trailing spaces
> 8:28:21 PM: docker.md:57: MD009 Trailing spaces

Can you correct?

**Comment by garysdevil** [comment]
> @garysdevil thanks for the updates, the build detected trailing spaces:
> 
> > 8:28:21 PM: docker.md:53: MD009 Trailing spaces
> > 8:28:21 PM: docker.md:57: MD009 Trailing spaces
> 
> Can you correct?

Corrected

**Review by bitschmidty — COMMENTED** [review]
Had an additional thought here, can we add a heading below [Building The Site Locally](https://github.com/bitcoinops/bitcoinops.github.io#building-the-site-locally) where we briefly summarize docker build and link to this doc?

**Comment by garysdevil** [comment]
> Building The Site Locally

It should be a good way

> Had an additional thought here, can we add a heading below [Building The Site Locally](https://github.com/bitcoinops/bitcoinops.github.io#building-the-site-locally) where we briefly summarize docker build and link to this doc?

It should be a good way, and it should be obvious so people can know there is another way to build the site.

**Comment by bitschmidty** [comment]
> > Had an additional thought here, can we add a heading below [Building The Site Locally](https://github.com/bitcoinops/bitcoinops.github.io#building-the-site-locally) where we briefly summarize docker build and link to this doc?
> 
> It should be a good way, and it should be obvious so people can know there is another way to build the site.

Feel free to add content and a link from the README to this new doc as part of this PR.

File: docker.md [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 1. Define the absolute parent path of repository bitcoinops.github.io .
     ```bash
-    repath=~/git/project/
+    repath=~/Desktop/
```
Comment (bitschmidty) [inline docker.md:9]: I dont have a preference on this default location either way. 

Personally, I have a ~/Desktop/ folder on my mac, although terminal did ask for permission.

Comment (garysdevil) [reply docker.md:9]: > > > Had an additional thought here, can we add a heading below [Building The Site Locally](https://github.com/bitcoinops/bitcoinops.github.io#building-the-site-locally) where we briefly summarize docker build and link to this doc?
> > 
> > 
> > It should be a good way, and it should be obvious so people can know there is another way to build the site.
> 
> Feel free to add content and a link from the README to this new doc as part of this PR.

There are two ways
1. Separate "building" detail instruction from README.md to a file, and only provide two building links in README.md. or
2. Just like I updated in the README.md file. It is a simple way.

File: README.md [hunk 4]
```diff
@@ -24,6 +24,8 @@ To build the site, you need to go through a one-time installation
 procedure that takes 15 to 30 minutes.  After that you can build the
 site an unlimited number of times with no extra work.
 
+**[You aleo can choose to build the site locally by docker, click here to get a guide.](./docker.md)**
```
Comment (bitschmidty) [inline README.md:27]:
```suggestion
**[You also can choose to build the site locally by docker, click here to get a guide.](./docker.md)**
```

Comment (bitschmidty) [reply README.md:27]: @garysdevil can you take a look at this small spelling correct?

File: docker.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    # Some of the dependencies can take a long time to install on some systems, so be patient.
+    docker run -d --name bitcoinops -p 4000:4000 -v ${repath}/bitcoinops.github.io:/root/bitcoinops.github.io -w /root/bitcoinops.github.io ruby:2.6.4-stretch /bin/bash -c "bundle install && make preview"
+    ```
+
+5. Output the bitcoinops container log
+    ```bash
+    docker logs -f bitcoinops
+    ```
+
+6. The bitcoinops container prints a log like this if building the site success.
```
Comment (bitschmidty) [inline docker.md:34]:
```suggestion
6. The bitcoinops container prints a log like this if building the site is successful.
```

File: docker.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
+[TOC]
+
+- Deploy docker document https://docs.docker.com/desktop/
```
Comment (bitschmidty) [inline docker.md:3]:
```suggestion
- Ensure Docker is installed and running (https://docs.docker.com/desktop/)
```

**Review by bitschmidty — COMMENTED** [review]
Ive run through this latest doc from scratch and just have few small wording suggestions @garysdevil if you are ok with these, I think we can merge!

File: docker.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    # Some of the dependencies can take a long time to install on some systems, so be patient.
+    docker run -d --name bitcoinops -p 4000:4000 -v ${repath}/bitcoinops.github.io:/root/bitcoinops.github.io -w /root/bitcoinops.github.io ruby:2.6.4-stretch /bin/bash -c "bundle install && make preview"
+    ```
+
+5. Output the bitcoinops container log
+    ```bash
+    docker logs -f bitcoinops
+    ```
+
+6. The bitcoinops container prints a log like this if building the site success.
```
Comment (bitschmidty) [inline docker.md:34]:
```suggestion
6. The bitcoinops container prints a log like this if building the site is successful.
```

**Pull request merged**